reconcile/utils/aws_api.py

```diff
@@ -2076,6 +2076,35 @@ def get_db_valid_upgrade_target(
             return versions[0]["ValidUpgradeTarget"]
         return []
 
+    def get_db_engine_version_status(
+        self,
+        account_name: str,
+        engine: str,
+        engine_version: str,
+        region_name: str | None = None,
+    ) -> str:
+        """
+        Get the version status of the database engine.
+        :param account_name: the name of the account in app-interface
+        :param engine: the database engine (ex. mysql, postgres)
+        :param engine_version: the database engine version
+        :param region_name: AWS region name for the resource, otherwise fallback to default
+
+        :return: https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/APIReference/API_DBEngineVersion.html
+        """
+        optional_kwargs = {}
+
+        if region_name:
+            optional_kwargs["region_name"] = region_name
+
+        rds = self._account_rds_client(account_name, **optional_kwargs)
+        response = rds.describe_db_engine_versions(
+            Engine=engine,
+            EngineVersion=engine_version,
+            IncludeAll=True,
+        )
+        return response["DBEngineVersions"][0]["Status"]
```

Comment:
`DBEngineVersions` can be empty if invalid `engine` or `engine_version` passed in, if we only need to know if the version is available or deprecated, can change this method to `is_db_engine_version_available` to return `bool` then do a simple query:


```python
        response = rds.describe_db_engine_versions(
            Engine=engine,
            EngineVersion=engine_version,
        )
        return len(response["DBEngineVersions"]) == 1
```